config/routes.rb

```diff
@@ -266,6 +266,7 @@
           get 'clone_template'
           get 'lock'
           get 'unlock'
+          post 'preview'
```

Comment:
I'd like to keep this PR minimal so it can still get into 1.9, adding API could be interesting but probably not that useful as in UI where you can modify template and preview it even before saving (with API you'd have to specify the whole template for preview and then do second query to save it)